src/solvers/dgsem_p4est/dg_3d.jl

```diff
@@ -21,34 +21,114 @@ function create_cache(mesh::P4estMesh{3}, equations, mortar_l2::LobattoLegendreM
 end
 
 
+#     index_to_start_step_3d(index::Symbol, index_range)
+#
+# Given a symbolic `index` and an `indexrange` (usually `eachnode(dg)`),
+# return an index value to begin a loop and index steps to update during a
+# loop. The resulting indices translate surface indices to volume indices.
+#
+# !!! warning
+#     This assumes that loops using the return values are ordered as
+#
+#     my_i, my_j, my_k = # begin values
+#     for j in ...
+#       for i in ...
+#         # do stuff with (i, j) and (my_i, my_j, my_k)
+#         my_[ijk] += my_[ijk]_step_i
+#       end
+#       my_[ijk] += my_[ijk]_step_j
+#     end
+@inline function index_to_start_step_3d(index::Symbol, index_range)
```

Comment:
Also, this function seems to return a 3-tuple. Which one is the start, which one is the step? If the latter two are the step, should we return those as a tuple?